nifi-extension-bundles/nifi-poi-bundle/nifi-poi-services/src/test/java/org/apache/nifi/excel/TestExcelRecordReader.java

```diff
@@ -318,17 +328,47 @@ void testPasswordProtected() throws Exception {
         assertEquals(DATA.length, records.size());
     }
 
+    @Test
+    void testPasswordProtectedOlderExcelFormat() throws Exception {
+        RecordSchema schema = getPasswordProtectedSchema();
+        ExcelRecordReaderConfiguration configuration = new ExcelRecordReaderConfiguration.Builder()
+                .withSchema(schema)
+                .withPassword(PASSWORD)
+                .withAvoidTempFiles(true)
+                .withInputFileType(InputFileType.XLS)
+                .build();
+
+        InputStream inputStream = new ByteArrayInputStream(PASSWORD_PROTECTED_OLDER_EXCEL.toByteArray());
+        ExcelRecordReader recordReader = new ExcelRecordReader(configuration, inputStream, logger);
+        List<Record> records = getRecords(recordReader, false, false);
+
+        assertEquals(DATA.length, records.size());
+    }
+
     @Test
     void testPasswordProtectedWithoutPassword() {
         RecordSchema schema = getPasswordProtectedSchema();
         ExcelRecordReaderConfiguration configuration = new ExcelRecordReaderConfiguration.Builder()
                 .withSchema(schema)
+                .withInputFileType(InputFileType.XLS)
                 .build();
 
         InputStream inputStream = new ByteArrayInputStream(PASSWORD_PROTECTED.toByteArray());
         assertThrows(Exception.class, () -> new ExcelRecordReader(configuration, inputStream, logger));
     }
 
+    @Test
+    void testPasswordProtectedOlderExcelWithoutPassword() {
+        RecordSchema schema = getPasswordProtectedSchema();
+        ExcelRecordReaderConfiguration configuration = new ExcelRecordReaderConfiguration.Builder()
+                .withSchema(schema)
+                .withInputFileType(InputFileType.XLS)
+                .build();
+
+        InputStream inputStream = new ByteArrayInputStream(PASSWORD_PROTECTED_OLDER_EXCEL.toByteArray());
+        assertThrows(Exception.class, () -> new ExcelRecordReader(configuration, inputStream, logger));
+    }
```

Comment:
Same as previous comment make this a Paramaterized test by combining this test with the one on lines 348-358